cms/djangoapps/contentstore/views/tests/test_block.py

```diff
@@ -3473,3 +3496,111 @@ def test_self_paced_item_visibility_state(self, store_type):
         # Check that in self paced course content has live state now
         xblock_info = self._get_xblock_info(chapter.location)
         self._verify_visibility_state(xblock_info, VisibilityState.live)
+
+
+@patch('xmodule.modulestore.split_mongo.caching_descriptor_system.CachingDescriptorSystem.applicable_aside_types',
+       lambda self, block: ['test_aside'])
+class TestUpdateFromSource(ModuleStoreTestCase):
+    """
+    Test update_from_source.
+    """
+
+    def setUp(self):
+        """
+        Set up the runtime for tests.
+        """
+        super().setUp()
+        key_store = DictKeyValueStore()
+        field_data = KvsFieldData(key_store)
+        self.runtime = TestRuntime(services={'field-data': field_data})
+
+    def create_source_block(self, course):
+        """
+        Create a chapter with all the fixings.
```

Comment:
You could also use `ToyCourseFactory` which has some content "with all the fixings" ready to go?

Reply:
That's out of pattern with the rest of the tests in this module, and I don't think it'd be as clear for these tests in particular, since I'll be testing specific attributes that would otherwise be implicitly created.

Reply:
OK, that's totally fine. Just didn't want you writing code if there was an existing thing you could use instead.